nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -204,8 +209,8 @@ let
 
     serviceConfig = {
       ExecStart = concatStringsSep " \\\n  " ([
-        "${pkgs.docker}/bin/docker run"
-        "--rm"
+        "${pkgs.docker}/bin/docker run -d"
```

Comment:
I'm not sure about daemonizing the container.  One of the nice things about keeping it in the foreground is that systemd can know when the container crashes and report it in `systemctl status` output, making these services behave somewhat more like normal systemd units.  What do you think?

Reply:
Seems like this switches process oversight from Systemd to Docker itself.

Reply:
> I'm not sure about daemonizing the container. One of the nice things about keeping it in the foreground is that systemd can know when the container crashes and report it in `systemctl status` output, making these services behave somewhat more like normal systemd units. What do you think?

```docker run``` process consumes memory without doing much and breaks logging. This also makes declared containers "special" because they behave in a way that is not obvious from docker container management perspective.

> Seems like this switches process oversight from Systemd to Docker itself. Is this because of Watchtower relying on docker being the process supervisor?

Yes